textproc/dblatex/Portfile

```diff
@@ -55,42 +55,39 @@ destroot.destdir    --root=${destroot} \
 # The mactex variant expects MacTeX to be installed
 # and installs dblatex's stylefiles to MacTeX's texmf (local)
 set dblatex.texmflocal ""
-set dblatex.mactex_bin ""
-
-variant mactex description {Allow dblatex to use a MacTeX installation instead of Macports texlive tools.
-To install the mactex variant /Library/TeX/texbin must be added to binpath
-in macports.conf} {
-    # First, check if MacTeX actually seems to be there…
-    set dblatex.mactex_candidates { \
-       "/Library/TeX/texbin" \
-       "/usr/texbin" \
-    }
-    foreach dir ${dblatex.mactex_candidates} {
-        if [file executable "${dir}/kpsewhich"] {
-            set dblatex.mactex_bin ${dir}
-            break
-        }
-    }
-    if { ${dblatex.mactex_bin} != "" } {
+
+variant mactex description {Allow dblatex to use MacTeX or another\
+    external TeXLive installation instead of MacPorts texlive tools.  To\
+    install the mactex variant the path to the TeX distribution's binary\
+    directory (for example /Library/TeX/texbin) must be added to binpath\
+    in macports.conf *before* installing this port.} {
+
+    # Check whether we have a TeXLive manager (which MacPorts doesn't
+    # provide).  We boldly assume that other basic TeXLive binaries
+    # like kpsewhich are then available also.
+    if {[llength [auto_execok tlmgr]]} {
         set dblatex.texmflocal \
-            [exec ${dblatex.mactex_bin}/kpsewhich --expand-var='\$TEXMFLOCAL']
+            [exec kpsewhich --expand-var='\$TEXMFLOCAL']
         regsub -all {'} ${dblatex.texmflocal} "" dblatex.texmflocal
     } else {
-        return -code error "Cannot find MacTeX installation; aborting"
+        return -code error "Cannot find MacTeX or external TeXLive\
+            installation; aborting.  Have you added the path to your\
+            TeX distribution's binary directory to binpath in\
+            macports.conf?"
     }
 
     depends_lib-delete \
                     port:texlive-latex-extra \
                     port:texlive-latex-recommended \
                     port:texlive-math-science
 
-    notes "The mactex variant will install styles to MacTeX's texmf-local
-        ${dblatex.texmflocal}
-    (which is outside macport's common directory structure).
-    For the mactex variant to work,
-    ${dblatex.mactex_bin} must also be added to binpath in macports.conf"
```

Comment:
Why did you delete this note? I suspect it's still needed.

Reply:
???  It's not removed but reworded, with improved formatting.

Reply:
The first part is reworded. OK, the second part was moved to `else return error ...` which probably makes sense. I saw it in variant description, but somehow missed the one in the error report.